cluster/mesos/docker/util.sh

```diff
@@ -0,0 +1,249 @@
+function test-build-release {
+  # Make a release
+  export KUBERNETES_CONTRIB=mesos
+  export KUBE_RELEASE_RUN_TESTS=N
+  "${KUBE_ROOT}/build/release.sh"
+}
+
+# Must ensure that the following ENV vars are set
+function detect-master {
+  #  echo "KUBE_MASTER: $KUBE_MASTER" 1>&2
+
+  docker_id=$(docker ps --filter="name=docker_apiserver" --quiet)
+  if [[ "${docker_id}" == *'\n'* ]]; then
+    echo "ERROR: Multiple API Servers running in docker" 1>&2
+    return 1
+  fi
+
+  details_json=$(docker inspect ${docker_id})
+  master_ip=$(jq '.[0].NetworkSettings.IPAddress' --raw-output <<< "$details_json")
+  master_port=$(jq '.[0].NetworkSettings.Ports[][].HostPort' --raw-output <<< "$details_json")
+
+  KUBE_MASTER_IP="${master_ip}:${master_port}"
+  KUBE_SERVER="http://${KUBE_MASTER_IP}"
+
+  echo "KUBE_MASTER_IP: $KUBE_MASTER_IP" 1>&2
+}
+
+# Get minion IP addresses and store in KUBE_MINION_IP_ADDRESSES[]
+# These Mesos slaves MAY host Kublets,
+# but might not have a Kublet running unless a kubernetes task has been scheduled on them.
+function detect-minions {
+  docker_ids=$(docker ps --filter="name=docker_mesosslave" --quiet)
+  while read -r docker_id; do
+
+    details_json=$(docker inspect ${docker_id})
+    minion_ip=$(jq '.[0].NetworkSettings.IPAddress' --raw-output <<< "$details_json")
+    minion_port=$(jq '.[0].NetworkSettings.Ports[][].HostPort' --raw-output <<< "$details_json")
+    #TODO: filter for 505* port
+
+    KUBE_MINION_IP_ADDRESSES+=("${minion_ip}")
+#    KUBE_MINION_IP_ADDRESSES+=("${minion_ip}:${minion_port}")
+
+  done <<< "$docker_ids"
+  echo "KUBE_MINION_IP_ADDRESSES: [${KUBE_MINION_IP_ADDRESSES[*]}]" 1>&2
+}
+
+# Verify prereqs on host machine
+function verify-prereqs {
+  echo "TODO: verify-prereqs" 1>&2
+  # Verify that docker, docker-compose, and jq exist
+  # Verify that all the right docker images exist:
+  # mesosphere/kubernetes-mesos-test, etc.
+}
+
+# Instantiate a kubernetes cluster
+function kube-up {
+  echo "Building Docker images" 1>&2
+  # TODO: version images (k8s version, git sha, and dirty state) to avoid re-building them every time.
+  "${provider_root}/km/build.sh"
+  "${provider_root}/mesos-slave/build.sh"
+  "${provider_root}/test/build.sh"
+
+  echo "Starting ${KUBERNETES_PROVIDER} cluster" 1>&2
+  pushd "${provider_root}" > /dev/null
+  docker-compose up -d
+  popd > /dev/null
+
+  detect-master
+  detect-minions
+
+  # await-health-check could be run locally, but it would require GNU timeout installed on mac...
+  # "${provider_root}/common/bin/await-health-check" -t=120 ${KUBE_SERVER}/healthz
+  cluster::mesos::docker::run_in_docker await-health-check -t=120 http://apiserver:8888/healthz
+
+  create-kubeconfig
+
+  echo "Deploying Addons" 1>&2
+  # deploy-addons could be run locally, but BSD base64 is has different behavior/arguments than GNU...
+  # "${provider_root}/deploy-addons.sh"
+  cluster::mesos::docker::run_in_docker "./cluster/${KUBERNETES_PROVIDER}/deploy-addons.sh"
```

Comment:
Running in docker when in a docker container already seems to break volume mounting: `Cannot start container... stat /go/src/github.com/GoogleCloudPlatform/kubernetes/./cluster/mesos/docker/deploy-addons.sh: no such file or directory`. I suspect we'll have to either detect being in a docker and run locally or mount KUBE_ROOT from a temp dir.